internal/logging/context.go

```diff
@@ -0,0 +1,279 @@
+package logging
```

Comment:
Please hold on commentary on this ... I did this while trying to make the whole logging system more fluent and less prone to multi-threaded problems... that said, this is kind ugly in that it duplicates a lot of the "odd" decisions that were in zerolog... I want to reduce this code down **a lot** before we merge.